frontend/src/scenes/feature-flags/featureFlagLogic.ts

```diff
@@ -416,18 +441,42 @@ export const featureFlagLogic = kea<featureFlagLogicType>([
             },
             saveFeatureFlag: async (updatedFlag: Partial<FeatureFlagType>) => {
                 const { created_at, id, ...flag } = updatedFlag
+
+                let preparedFlag = flag
+                // Add payloads mapped to keys in multivariate
+                if (preparedFlag.filters?.multivariate) {
```

Comment:
I see the awkwardness here, do you have an alternative schema proposal?

I'm a bit unwilling to design the API based on what's convenient for the frontend in this case, because it affects not just other API clients, but all client libraries as well.

However, if there's something that works easy-enough for both, would be more than happy.

The problem with putting payloads inside multivariates is that it only works for the multivariate case, and every where else needs to figure out where to get payloads from.

---

Hmm, one potential way of making things easier on the frontend and keeping a reasonable API might be to convert the `payloads` dict into an array instead. In this case, it's not keyed to the variant name, so can be updated independently, and isn't ambiguous on the backend either.

hmm, although, that's pretty shite for the libraries, as they can't match by match value anymore. Hm. Well, we'll just move the transformation from here to those libraries instead.

Reply:
Actually, when a variant value changes, we know both the old and new values, so why can't we just update the payload at the same time as well?

That gets rid of this awkward mapping, and generates the correct payload from the get-go, doesn't it?

Reply:
> Actually, when a variant value changes, we know both the old and new values, so why can't we just update the payload at the same time as well?

For the case where we edit, yes this will work. But when the form is brand new and if someone starts filling in a payload first, we need to store by index and then convert to by key. Then we have multiple modes of handling payloads

Reply:
ugh, makes sense 👍